lib/stack.go

```diff
@@ -0,0 +1,36 @@
+package lib
+
+// Stack type
+type Stack []string
+
+// IsEmpty check if stack is empty
+func (st *Stack) IsEmpty() bool {
+	return len(*st) == 0
+}
+
+// Push a new value onto the stack
+func (st *Stack) Push(str string) {
+	*st = append(*st, str) //Simply append the new value to the end of the stack
+}
+
+// Pop Remove top element of stack. Return false if stack is empty.
+func (st *Stack) Pop() bool {
```

Comment:
This implementation pops without returning the popped element. Not really ideal...

Reply:
Fixed. Combined the functionality of `Pop` and `Top`  into a one function `Pop`.